docs/book/src/reference/admission-webhook.md

```diff
@@ -0,0 +1,38 @@
+# Admission Webhook
+
+Admission webhooks are HTTP callbacks that receive admission requests, process them and return admission responses.
+There are two types of admission webhooks: mutating admission webhook and validating admission webhook.
+With mutating admission webhooks, you may change the request object before it is stored (e.g. for implementing defaulting of fields)
+With validating admission webhooks, you may not change the request, but you can reject it (e.g. for implementing validation of the request).
+
+### Why Admission Webhooks are Important
+
+Admission webhooks are the mechanism to enable kubernetes extensibility through CRD.
+- Mutating admission webhook is the only way to do defaulting for CRDs.
+- Validating admission webhook allows for more complex validation than pure schema-based validation.
+e.g. cross-field validation or cross-object validation.
```

Comment:
we discourage cross-object validation. But we can say, validating with an external source etc.

Reply:
yeah, you should almost never do cross-object validation.